temporalio/activity.py

```diff
@@ -64,15 +73,19 @@ def defn(
     Args:
         fn: The function to decorate.
         name: Name to use for the activity. Defaults to function ``__name__``.
+            This cannot be set if dynamic is set.
         no_thread_cancel_exception: If set to true, an exception will not be
             raised in synchronous, threaded activities upon cancellation.
+        dynamic: If true, this activity will be dynamic. Dynamic activities have
+            to accept a single 'Sequence[RawValue]' parameter. This cannot be
+            set to true if name is present.
```

Comment:
This doesn't appear to actually be enforced, just ignored. Would be nice to actually enforce (or better, make unrepresentable if reasonable)

Reply:
It is unrepresentable with the overloads if you use a type checker which everyone should (I need to fix this for the workflow side). Right now, name is just ignored when dynamic is present. If we want a runtime check, I can add it.